lib/models/account-manager.js

```diff
@@ -345,18 +345,19 @@ class AccountManager {
       username: userData.username,
       email: userData.email,
       name: userData.name,
+      externalWebId: userData.externalWebId,
       webId: userData.webid || userData.webId ||
         this.accountWebIdFor(userData.username)
     }
 
-    if (userConfig.webId && !userConfig.username) {
-      userConfig.username = this.usernameFromWebId(userConfig.webId)
-    }
-
     if (!userConfig.webId && !userConfig.username) {
       throw new Error('Username or web id is required')
     }
 
+    if (userConfig.webId && !userConfig.username) {
+      userConfig.username = this.usernameFromWebId(userConfig.webId)
+    }
+
```

Comment:
@dmitrizagidulin MIght want to group those two `if`s in an `if (!userConfig.username)`?